### PR TITLE
B-19176-INT SC upload orders

### PR DESCRIPTION
**Pull request by paulstonebraker**
## Summary

Services Counselors can now upload documents for customers orders.

### How to test

1. Access the application as a Services Counselor.
2. Search for a customer and create a new move.
3. Once on the move details page, click View and edit orders button
4. Because no orders documents are present, you should see the Add orders button. Clicking it should expand the upload orders section. Upload a file or files; when you finish click done and confirm the document viewer refreshes with your uploaded document(s).
5. Navigate away and return, and confirm your uploads persist, and the ability to upload files is no longer there (uploading amended orders will come later)

### CODE REVIEW

**Comment by landan-parker-0** [comment]
It works for me.
When I click 'done' before the upload is complete, it unmounts the previewer upload while it's processing, and does not proceed with the upload.
I think the done button should be disabled until the upload is completed.

**Comment by paulstonebraker** [comment]
> It works for me. When I click 'done' before the upload is complete, it unmounts the previewer upload while it's processing, and does not proceed with the upload. I think the done button should be disabled until the upload is completed.

Good find, and I agree. Awaiting confirmation from PO but after discussion with coach clayton lets proceed as if this is not in the scope of this work and will be addressed as part of file deletion.

**Review by pambecker — APPROVED** [review]

**Review by landan-parker-0 — APPROVED** [review]
Works correctly according to criteria

File: pkg/handlers/ghcapi/uploads.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+)
+
+type CreateUploadHandler struct {
+	handlers.HandlerConfig
+}
+
+func (h CreateUploadHandler) Handle(params uploadop.CreateUploadParams) middleware.Responder {
+	return h.AuditableAppContextFromRequestWithErrors(params.HTTPRequest,
+		func(appCtx appcontext.AppContext) (middleware.Responder, error) {
+			rollbackErr := fmt.Errorf("error creating upload")
```
Comment (danieljordan-caci) [inline pkg/handlers/ghcapi/uploads.go:26]: We don't want to use `fmt.Errorf` really at all. Using `appCtx` to log errors is preferred.

Can you instead do:
```
appCtx.Logger().Error("error creating upload")
```

File: swagger-def/ghc.yaml [hunk 2]
```diff
@@ -3439,6 +3440,44 @@ paths:
...
+      consumes:
+        - multipart/form-data
+      produces:
+        - application/json
+      parameters:
+        - in: query
+          name: documentId
+          type: string
+          format: uuid
+          required: false
```
Comment (danieljordan-caci) [inline swagger-def/ghc.yaml:3459]: You can just remove this `required: false` line if ya wanna

**Review by danieljordan-caci — CHANGES_REQUESTED** [review]
Just a couple little things, but this is great work.

I did find a break in the code when reviewing this and I tagged @joeydoyecaci in his PR where he made those changes. Just make sure we get those changes back into `integrationTesting` before merging the `main` PR for that BL item. Thanks!

**Review by pambecker — APPROVED** [review]

**Review by danieljordan-caci — APPROVED** [review]
Send it! LGTM!

**Pull request merged**